### DISCUSSION TITLE
How to put the project content in subfolder?

### DISCUSSION THREAD

**Discussion by josedusol** [quarto-dev/quarto-cli#3759] [Q&A] [answered]
Quarto, just like bookdown, forces me to have a non empty index.qmd file in the root folder, even if i'm only interested in pdf output. I can't just remove index.qmd from the `chapters` attribute in yaml file because i get 

> ERROR: Book contents must include a home page (e.g. index.md)

I would like to put all my book content in a subfolder. Is this possible?

**Comment by cscheid** [comment, marked as answer]
It's not currently possible.

**Reply by josedusol** [reply]
Thanks for the info.

But, is there any workaround?. For example, in bookdown one can put an empty `index.Rmd` file in project root, and exclude it from loaded Rmd files in configuration file. Then, we can have e.g. `subfolder/introduction.Rmd` as main file, where we declare first section as:

```
# Introduction {#index}
```
This trick es enough for bookdown, the index file for html output is created from `subfolder/introduction.Rmd`. And it is not used for other output formats.

Unfortunately, this does not work in Quarto, maybe because `{#label}` doesn't set the file name to `label` as in bookdown, but i'm not sure.

**Reply by josedusol** [reply]
Or there is some way to change the name of index.qmd?

**Reply by cscheid** [reply]
I'm not aware of a workaround. Is there a specific reason you don't want to have an `index.qmd` file?

**Reply by josedusol** [reply]
Personally, i would like to name my project files with meaningful names representing the structure of my book. For example: `preface.qmd`, `introduction.qmd`, `whatever.qmd`, etc. But for some reason something named `index.qmd` needs to be there. So, people typically uses `index.qmd` to put the preface or introduction. This is not big deal, but is ugly (ok, i imagine this have priority 0). The same goes for bookdown.

I found quarto useful because it allows me to abstract away from concrete formats (HTML, PDF, ETC), is the tool responsibility to generate the concrete formats. The obligation to have something with name `index` is HTML specific. I hope the tool could just create this file for the HTML output behind the scenes.

That said, i'm completely ignorant of the inner workings, surely there are reasons i don't know.

Anyway, thanks for the chat Carlos. Keep up the good work and happy new year!.

**Reply by cscheid** [reply]
> The obligation to have something with name index is HTML specific.

It also happens to bookdown and quarto specific (in other words, we chose this). We can argue about aesthetics, but `index` is a commonly-accepted file name for "default", so we use it.
